# Практическая работа №3: Обработка выборочных данных. Нахождение интервальных оценок параметров распределения. Проверка статистической гипотезы о нормальном распределении

Выполнили студентки гр. 2384 Лавренова Юлия и Соц Екатерина. Вариант №32

## Цель работы
Получение практических навыков вычисления интервальных статистических оценок параметров распределения выборочных данных и проверки «справедливости» статистических гипотез.

## Основные теоретические положения
**Интервальной** называется статистическая оценка, которая задается двумя числами - концами интервала (в котором может находиться значение параметра $\Theta$)

**Тоностью $\Theta^*$** называется значение $\delta$, удовлетворяющее неравенству: $\delta > |\Theta^* - \Theta|$

**Надежностью (доверительной вероятностью)** оценки называется вероятность $\gamma$, с которой выполняется неравество, т е $\gamma = P \{\delta > |\Theta^* - \Theta|\}$

**Интервальные статистические оценки параметров распределения случайной величины:** $\gamma = P \{\Theta^* - \delta < \Theta < \Theta^* + \delta\}$

**Доверительный интервал:** $(\Theta^* - \delta ;\Theta^* + \delta )$

При двусторонней оценке задаются обе границы доверительного интервала, зависящие от доверительной вероятность $\gamma$: $P \{\beta_1(\gamma)<\Theta<\beta_2(\gamma) \} = p$

**Уровень значимости** $q=1-\gamma$

**Доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении при заданном объеме выборки**: $P(\bar x_в - t_\gamma S / \sqrt N,\bar x_в + t_\gamma S / \sqrt N,)$, где
 - $\bar{x_в}$ — выборочное среднее
- $S$ — исправленное СКВО
 - $N$ — объем выборки
- $t_{\gamma}$ — квантиль распределения Стьюдента для доверительной вероятности $\gamma$

**Проверка гипотезы о нормальности (критерий Пирсона $\chi^2$)**

**Нулевая гипотиза $H_0$:** Случайная величина имеет нормальное распределение.

**Конкурирующая гипотеза $H_1$:** Распределение не является нормальным.

**Критерий Пирсона $\chi^2$** для проверки нормальности:
$$\chi^2_{набл} = \sum_{i=1}^{k} \frac{(n_i - n^{\prime}_i)^2}{n^{\prime}_i}, \quad n^{\prime}_i = n \cdot p_i$$
Число степеней свободы:
$$df = k - 3$$, k - число интервалов

Гипотеза $H_0$ принимается, если
$$\chi^2_{набл} < \chi^2_{крит}$$
, иначе - отвергается

## Постановка задачи
Для заданной надежности определить (на основании выборочных данных и результатов выполнения практической работы №2) границы доверительных интервалов для математического ожидания и среднеквадратичного отклонения случайной величины. Проверить гипотезу о нормальном распределении исследуемой случайной величины с помощью критерия Пирсона $\chi^2$. Дать содержательную интерпретацию полученным результатам.

## Выполнение работы

Наработки из первой практической работы:

In [ ]:
import pandas as pd

data = pd.read_csv("sample_wine_data.csv")

In [ ]:
import numpy as np

def group_array(data):
  # ранжированный
  df_range = np.sort(data)

  # вариационный
  df_unique = np.unique(df_range)
  df_var = {value: np.sum(df_range == value) for value in df_unique}

  # интервальный

  N = 114
  k = round(1 + 3.322*np.log10(N))
  h = (max(df_unique) - min(df_unique)) / k
  bins = np.linspace(min(df_unique), max(df_unique), k)
  m_i, edges = np.histogram(data, bins=bins)

  x_i_mid = (edges[:-1] + edges[1:]) / 2
  m_i_rel = m_i / N
  m_i_cum = np.cumsum(m_i)
  m_i_rel_cum = np.cumsum(m_i_rel)
  interval_series = pd.DataFrame({
      "Интервал": [f"{round(edges[i],2)} – {round(edges[i+1],2)}" for i in range(len(m_i))],
      "Абсолютная частота": m_i,
      "Относительная частота": m_i / N
  })
  table = pd.DataFrame({
      "i": range(1, len(m_i)+1),
      "$[x_i, x_{i+1})$": [f"[{edges[i]:.2f}, {edges[i+1]:.2f})" for i in range(len(m_i))],
      "$\\tilde{x}_i$": np.round(x_i_mid, 3),
      "$m_i$": m_i,
      "$\\tilde{m}_i$": np.round(m_i_rel, 4),
      "$m_{i}^{нак}$": m_i_cum,
      "$\\tilde{m}_{i}^{нак}$": np.round(m_i_rel_cum, 4)
  })


  sum_row = pd.DataFrame({
      "i": ["Σ"],
      "$[x_i, x_{i+1})$": ["-"],
      "$\\tilde{x}_i$": ["-"],
      "$m_i$": [m_i.sum()],
      "$\\tilde{m}_i$": [m_i_rel.sum()],
      "$m_{i}^{нак}$": ["-"],
      "$\\tilde{m}_{i}^{нак}$": ["-"]
  })

  table = pd.concat([table, sum_row], ignore_index=True)

  return df_range, df_var, interval_series, table
data_ph = np.array(data["pH"])
ph_range, ph_var, ph_interval, ph_table = group_array(data_ph)

from IPython.display import display, Markdown

display(Markdown(ph_table.to_markdown(index=False)))

data_sulphates = np.array(data["sulphates"])
sulphates_range, sulphates_var, sulphates_interval, sulphates_table = group_array(data_sulphates)

display(Markdown(sulphates_table.to_markdown(index=False)))

| i   | $[x_i, x_{i+1})$   | $\tilde{x}_i$   |   $m_i$ |   $\tilde{m}_i$ | $m_{i}^{нак}$   | $\tilde{m}_{i}^{нак}$   |
|:----|:-------------------|:----------------|--------:|----------------:|:----------------|:------------------------|
| 1   | [2.86, 3.01)       | 2.934           |       5 |          0.0439 | 5               | 0.0439                  |
| 2   | [3.01, 3.16)       | 3.083           |      12 |          0.1053 | 17              | 0.1491                  |
| 3   | [3.16, 3.31)       | 3.231           |      38 |          0.3333 | 55              | 0.4825                  |
| 4   | [3.31, 3.45)       | 3.38            |      41 |          0.3596 | 96              | 0.8421                  |
| 5   | [3.45, 3.60)       | 3.529           |      13 |          0.114  | 109             | 0.9561                  |
| 6   | [3.60, 3.75)       | 3.677           |       3 |          0.0263 | 112             | 0.9825                  |
| 7   | [3.75, 3.90)       | 3.826           |       2 |          0.0175 | 114             | 1.0                     |
| Σ   | -                  | -               |     114 |          1      | -               | -                       |

| i   | $[x_i, x_{i+1})$   | $\tilde{x}_i$   |   $m_i$ |   $\tilde{m}_i$ | $m_{i}^{нак}$   | $\tilde{m}_{i}^{нак}$   |
|:----|:-------------------|:----------------|--------:|----------------:|:----------------|:------------------------|
| 1   | [0.44, 0.61)       | 0.524           |      53 |          0.4649 | 53              | 0.4649                  |
| 2   | [0.61, 0.78)       | 0.693           |      42 |          0.3684 | 95              | 0.8333                  |
| 3   | [0.78, 0.95)       | 0.861           |      13 |          0.114  | 108             | 0.9474                  |
| 4   | [0.95, 1.11)       | 1.03            |       1 |          0.0088 | 109             | 0.9561                  |
| 5   | [1.11, 1.28)       | 1.199           |       2 |          0.0175 | 111             | 0.9737                  |
| 6   | [1.28, 1.45)       | 1.367           |       2 |          0.0175 | 113             | 0.9912                  |
| 7   | [1.45, 1.62)       | 1.536           |       1 |          0.0088 | 114             | 1.0                     |
| Σ   | -                  | -               |     114 |          1      | -               | -                       |

In [ ]:
# Извлекаем данные из ph_table
ph_data = ph_table.iloc[:-1].copy()

ph_data['i'] = ph_data['i'].astype(int)
ph_data['$\\tilde{x}_i$'] = ph_data['$\\tilde{x}_i$'].astype(float)
ph_data['$m_i$'] = ph_data['$m_i$'].astype(int)

print("Исходные данные для pH:")
display(Markdown(ph_data[['i', '$\\tilde{x}_i$', '$m_i$']].to_markdown(index=False)))

Исходные данные для pH:


|   i |   $\tilde{x}_i$ |   $m_i$ |
|----:|----------------:|--------:|
|   1 |           2.934 |       5 |
|   2 |           3.083 |      12 |
|   3 |           3.231 |      38 |
|   4 |           3.38  |      41 |
|   5 |           3.529 |      13 |
|   6 |           3.677 |       3 |
|   7 |           3.826 |       2 |

In [ ]:
def calculate_conditional_variants(data_table, feature_name):

    df = data_table.copy()

    # Определяем шаг интервала h
    # Берем разницу между первыми двумя серединами интервалов
    x_mid = df['$\\tilde{x}_i$'].values
    h = round(x_mid[1] - x_mid[0], 3)

    # Выбираем условный нуль C (середина интервала с максимальной частотой)
    max_freq_idx = df['$m_i$'].idxmax()
    C = df.loc[max_freq_idx, '$\\tilde{x}_i$']

    print(f"ВЫЧИСЛЕНИЕ УСЛОВНЫХ ВАРИАНТ ДЛЯ ПРИЗНАКА: {feature_name}")
    print(f"Шаг интервала h = {h}")
    print(f"Условный нуль C = {C} (середина интервала с максимальной частотой)")

    # условные варианты u_i
    df['$u_i$'] = ((df['$\\tilde{x}_i$'] - C) / h).round(3)

    df['$n_i u_i$'] = (df['$m_i$'] * df['$u_i$']).round(3)
    df['$n_i u_i^2$'] = (df['$m_i$'] * df['$u_i$']**2).round(3)
    df['$n_i u_i^3$'] = (df['$m_i$'] * df['$u_i$']**3).round(3)
    df['$n_i u_i^4$'] = (df['$m_i$'] * df['$u_i$']**4).round(3)
    df['$n_i (u_i+1)^4$'] = (df['$m_i$'] * (df['$u_i$'] + 1)**4).round(3)

    result_table = df[['i', '$\\tilde{x}_i$', '$m_i$', '$u_i$',
                       '$n_i u_i$', '$n_i u_i^2$', '$n_i u_i^3$',
                       '$n_i u_i^4$', '$n_i (u_i+1)^4$']].copy()

    sums = pd.DataFrame({
        'i': ['Σ'],
        '$\\tilde{x}_i$': ['-'],
        '$m_i$': [df['$m_i$'].sum()],
        '$u_i$': ['-'],
        '$n_i u_i$': [result_table['$n_i u_i$'].sum()],
        '$n_i u_i^2$': [result_table['$n_i u_i^2$'].sum()],
        '$n_i u_i^3$': [result_table['$n_i u_i^3$'].sum()],
        '$n_i u_i^4$': [result_table['$n_i u_i^4$'].sum()],
        '$n_i (u_i+1)^4$': [result_table['$n_i (u_i+1)^4$'].sum()]
    })

    final_table = pd.concat([result_table, sums], ignore_index=True)

    return final_table, h, C, df['$m_i$'].sum()

# для pH
ph_conditional_table, ph_h, ph_C, ph_n = calculate_conditional_variants(ph_data, "pH")

display(Markdown(ph_conditional_table.to_markdown(index=False)))

ВЫЧИСЛЕНИЕ УСЛОВНЫХ ВАРИАНТ ДЛЯ ПРИЗНАКА: pH
Шаг интервала h = 0.149
Условный нуль C = 3.38 (середина интервала с максимальной частотой)


| i   | $\tilde{x}_i$   |   $m_i$ | $u_i$   |   $n_i u_i$ |   $n_i u_i^2$ |   $n_i u_i^3$ |   $n_i u_i^4$ |   $n_i (u_i+1)^4$ |
|:----|:----------------|--------:|:--------|------------:|--------------:|--------------:|--------------:|------------------:|
| 1   | 2.934           |       5 | -2.993  |     -14.965 |        44.79  |      -134.057 |       401.233 |            78.886 |
| 2   | 3.083           |      12 | -1.993  |     -23.916 |        47.665 |       -94.996 |       189.326 |            11.668 |
| 3   | 3.231           |      38 | -1.0    |     -38     |        38     |       -38     |        38     |             0     |
| 4   | 3.38            |      41 | 0.0     |       0     |         0     |         0     |         0     |            41     |
| 5   | 3.529           |      13 | 1.0     |      13     |        13     |        13     |        13     |           208     |
| 6   | 3.677           |       3 | 1.993   |       5.979 |        11.916 |        23.749 |        47.332 |           240.74  |
| 7   | 3.826           |       2 | 2.993   |       5.986 |        17.916 |        53.623 |       160.493 |           508.425 |
| Σ   | -               |     114 | -       |     -51.916 |       173.287 |      -176.681 |       849.384 |          1088.72  |

In [ ]:
def check_calc(table, n):
    # Σ n_j (u_j + 1)^4 = Σ n_j u_j^4 + 4Σ n_j u_j^3 + 6Σ n_j u_j^2 + 4Σ n_j u_j + n

    sum_ni_ui = table[table['i'] != 'Σ']['$n_i u_i$'].sum()
    sum_ni_ui2 = table[table['i'] != 'Σ']['$n_i u_i^2$'].sum()
    sum_ni_ui3 = table[table['i'] != 'Σ']['$n_i u_i^3$'].sum()
    sum_ni_ui4 = table[table['i'] != 'Σ']['$n_i u_i^4$'].sum()
    sum_ni_ui_plus1_4 = table[table['i'] != 'Σ']['$n_i (u_i+1)^4$'].sum()

    # Вычисляем правую часть формулы
    right_side = sum_ni_ui4 + 4*sum_ni_ui3 + 6*sum_ni_ui2 + 4*sum_ni_ui + n

    print(f"Σ n_i u_i = {sum_ni_ui:.3f}")
    print(f"Σ n_i u_i^2 = {sum_ni_ui2:.3f}")
    print(f"Σ n_i u_i^3 = {sum_ni_ui3:.3f}")
    print(f"Σ n_i u_i^4 = {sum_ni_ui4:.3f}")
    print(f"Σ n_i (u_i+1)^4 = {sum_ni_ui_plus1_4:.3f}")
    print(f"n = {n}")
    print(f"Левая часть: Σ n_i (u_i+1)^4 = {sum_ni_ui_plus1_4:.3f}")
    print(f"Правая часть: Σ n_i u_i^4 + 4Σ n_i u_i^3 + 6Σ n_i u_i^2 + 4Σ n_i u_i + n =")
    print(f"  {sum_ni_ui4:.3f} + 4*{sum_ni_ui3:.3f} + 6*{sum_ni_ui2:.3f} + 4*{sum_ni_ui:.3f} + {n} =")
    print(f"  {sum_ni_ui4:.3f} + {4*sum_ni_ui3:.3f} + {6*sum_ni_ui2:.3f} + {4*sum_ni_ui:.3f} + {n} =")
    print(f"  {right_side:.3f}")

    if abs(right_side - sum_ni_ui_plus1_4) < 0.01:
        print("\n Равенство выполняется")
        print(f"  Разница: {abs(right_side - sum_ni_ui_plus1_4):.6f}")
    else:
        print("\n Равенство не выполняется")
        print(f"  Разница: {abs(right_side - sum_ni_ui_plus1_4):.6f}")

check_calc(ph_conditional_table, ph_n)

Σ n_i u_i = -51.916
Σ n_i u_i^2 = 173.287
Σ n_i u_i^3 = -176.681
Σ n_i u_i^4 = 849.384
Σ n_i (u_i+1)^4 = 1088.719
n = 114
Левая часть: Σ n_i (u_i+1)^4 = 1088.719
Правая часть: Σ n_i u_i^4 + 4Σ n_i u_i^3 + 6Σ n_i u_i^2 + 4Σ n_i u_i + n =
  849.384 + 4*-176.681 + 6*173.287 + 4*-51.916 + 114 =
  849.384 + -706.724 + 1039.722 + -207.664 + 114 =
  1088.718

 Равенство выполняется
  Разница: 0.001000


In [ ]:
# Извлекаем данные для sulphates
sulphates_data = sulphates_table.iloc[:-1].copy()
sulphates_data['i'] = sulphates_data['i'].astype(int)
sulphates_data['$\\tilde{x}_i$'] = sulphates_data['$\\tilde{x}_i$'].astype(float)
sulphates_data['$m_i$'] = sulphates_data['$m_i$'].astype(int)

# Вычисляем условные варианты для sulphates
sulphates_conditional_table, sulphates_h, sulphates_C, sulphates_n = calculate_conditional_variants(sulphates_data, "SULPHATES")

display(Markdown(sulphates_conditional_table.to_markdown(index=False)))

check_calc(sulphates_conditional_table, sulphates_n)

ВЫЧИСЛЕНИЕ УСЛОВНЫХ ВАРИАНТ ДЛЯ ПРИЗНАКА: SULPHATES
Шаг интервала h = 0.169
Условный нуль C = 0.524 (середина интервала с максимальной частотой)


| i   | $\tilde{x}_i$   |   $m_i$ | $u_i$   |   $n_i u_i$ |   $n_i u_i^2$ |   $n_i u_i^3$ |   $n_i u_i^4$ |   $n_i (u_i+1)^4$ |
|:----|:----------------|--------:|:--------|------------:|--------------:|--------------:|--------------:|------------------:|
| 1   | 0.524           |      53 | 0.0     |       0     |         0     |         0     |         0     |            53     |
| 2   | 0.693           |      42 | 1.0     |      42     |        42     |        42     |        42     |           672     |
| 3   | 0.861           |      13 | 1.994   |      25.922 |        51.688 |       103.067 |       205.515 |          1044.6   |
| 4   | 1.03            |       1 | 2.994   |       2.994 |         8.964 |        26.838 |        80.354 |           254.467 |
| 5   | 1.199           |       2 | 3.994   |       7.988 |        31.904 |       127.425 |       508.935 |          1244.01  |
| 6   | 1.367           |       2 | 4.988   |       9.976 |        49.76  |       248.204 |      1238.04  |          2571.33  |
| 7   | 1.536           |       1 | 5.988   |       5.988 |        35.856 |       214.707 |      1285.66  |          2384.58  |
| Σ   | -               |     114 | -       |      94.868 |       220.172 |       762.241 |      3360.51  |          8223.98  |

Σ n_i u_i = 94.868
Σ n_i u_i^2 = 220.172
Σ n_i u_i^3 = 762.241
Σ n_i u_i^4 = 3360.510
Σ n_i (u_i+1)^4 = 8223.983
n = 114
Левая часть: Σ n_i (u_i+1)^4 = 8223.983
Правая часть: Σ n_i u_i^4 + 4Σ n_i u_i^3 + 6Σ n_i u_i^2 + 4Σ n_i u_i + n =
  3360.510 + 4*762.241 + 6*220.172 + 4*94.868 + 114 =
  3360.510 + 3048.964 + 1321.032 + 379.472 + 114 =
  8223.978

 Равенство выполняется
  Разница: 0.005000


In [ ]:
results = {
    'ph': {
        'table': ph_conditional_table,
        'h': ph_h,
        'C': ph_C,
        'n': ph_n,
        'x_mid': ph_data['$\\tilde{x}_i$'].values,
        'freq': ph_data['$m_i$'].values
    },
    'sulphates': {
        'table': sulphates_conditional_table,
        'h': sulphates_h,
        'C': sulphates_C,
        'n': sulphates_n,
        'x_mid': sulphates_data['$\\tilde{x}_i$'].values,
        'freq': sulphates_data['$m_i$'].values
    }
}


In [ ]:
def calculate_moments(results, feature_name):
    print(f"ВЫЧИСЛЕНИЕ МОМЕНТОВ ДЛЯ ПРИЗНАКА: {feature_name.upper()}")

    table = results[feature_name]['table']
    n = results[feature_name]['n']
    h = results[feature_name]['h']
    C = results[feature_name]['C']

    sum_ni_ui = table[table['i'] != 'Σ']['$n_i u_i$'].sum()
    sum_ni_ui2 = table[table['i'] != 'Σ']['$n_i u_i^2$'].sum()
    sum_ni_ui3 = table[table['i'] != 'Σ']['$n_i u_i^3$'].sum()
    sum_ni_ui4 = table[table['i'] != 'Σ']['$n_i u_i^4$'].sum()

    # Условные эмпирические моменты M*_k"
    M1_star = sum_ni_ui / n
    M2_star = sum_ni_ui2 / n
    M3_star = sum_ni_ui3 / n
    M4_star = sum_ni_ui4 / n

    # Центральные эмпирические моменты m_k
    # m1 всегда равен 0
    m1 = 0
    m2 = h**2 * (M2_star - M1_star**2)
    m3 = h**3 * (M3_star - 3*M1_star*M2_star + 2*M1_star**3)
    m4 = h**4 * (M4_star - 4*M1_star*M3_star + 6*M1_star**2 * M2_star - 3*M1_star**4)

    moments_table = pd.DataFrame({
        "k": [1, 2, 3, 4],
        "$\\bar{M}^*_k$": [f"{M1_star:.4f}", f"{M2_star:.4f}", f"{M3_star:.4f}", f"{M4_star:.4f}"],
        "$\\bar{m}_k$": [f"{m1}", f"{m2:.4f}", f"{m3:.4f}", f"{m4:.4f}"]
    })

    display(Markdown(moments_table.to_markdown(index=False)))

    moments_results = {
        'M1_star': M1_star,
        'M2_star': M2_star,
        'M3_star': M3_star,
        'M4_star': M4_star,
        'm1': m1,
        'm2': m2,
        'm3': m3,
        'm4': m4,
        'h': h,
        'C': C,
        'n': n
    }

    return moments_results

ph_moments = calculate_moments(results, 'ph')
sulphates_moments = calculate_moments(results, 'sulphates')

ВЫЧИСЛЕНИЕ МОМЕНТОВ ДЛЯ ПРИЗНАКА: PH


|   k |   $\bar{M}^*_k$ |   $\bar{m}_k$ |
|----:|----------------:|--------------:|
|   1 |         -0.4554 |        0      |
|   2 |          1.5201 |        0.0291 |
|   3 |         -1.5498 |        0.0011 |
|   4 |          7.4507 |        0.0031 |

ВЫЧИСЛЕНИЕ МОМЕНТОВ ДЛЯ ПРИЗНАКА: SULPHATES


|   k |   $\bar{M}^*_k$ |   $\bar{m}_k$ |
|----:|----------------:|--------------:|
|   1 |          0.8322 |        0      |
|   2 |          1.9313 |        0.0354 |
|   3 |          6.6863 |        0.0146 |
|   4 |         29.4782 |        0.0113 |

In [ ]:
def calculate_sample_characteristics(moments_results, raw_data=None, feature_name=""):
    print(f"\n    ВЫЧИСЛЕНИЕ ВЫБОРОЧНЫХ ХАРАКТЕРИСТИК ДЛЯ ПРИЗНАКА: {feature_name}\n")

    C = moments_results['C']
    h = moments_results['h']
    M1_star = moments_results['M1_star']
    m2 = moments_results['m2']
    n = moments_results['n']

    print("Вычисление через условные варианты")
    x_bar_cond = C + h * M1_star
    D_cond = m2
    sigma_cond = np.sqrt(D_cond)

    print(f"    Выборочное среднее через условные варианты: {x_bar_cond:.4f}")
    print(f"    Выборочная дисперсия через условные варианты: {m2:.4f}")
    print(f"    Выборочное СКО: {sigma_cond:.4f}")

    if raw_data is not None:
        print("Вычисление по стандартной формуле")
        x_bar_std = np.mean(raw_data)
        D_std = np.var(raw_data, ddof=0)
        sigma_std = np.std(raw_data, ddof=0)

        print(f"    Выборочное среднее по стандартной формуле: {x_bar_std:.4f}")
        print(f"    Выборочная дисперсия по стандартной формуле: {D_std:.4f}")
        print(f"    Выборочное СКО по стандартной формуле: {sigma_std:.4f}")

        print("\n   СРАВНЕНИЕ РЕЗУЛЬТАТОВ\n")
        print("Сравнение выборочного среднего:")
        diff_mean = abs(x_bar_cond - x_bar_std)
        if diff_mean < 0.01:
            print(f"  РЕЗУЛЬТАТЫ СОВПАДАЮТ (разница: {diff_mean:.6f})")
        else:
            print(f"  РАСХОЖДЕНИЕ (разница: {diff_mean:.6f})")

        print("Сравнение дисперсии:")
        diff_var = abs(D_cond - D_std)
        if diff_var < 0.01:
            print(f"  РЕЗУЛЬТАТЫ СОВПАДАЮТ (разница: {diff_var:.6f})")
        else:
            print(f"  РАСХОЖДЕНИЕ (разница: {diff_var:.6f})")

        print("Сравнение СКО:")
        diff_std = abs(sigma_cond - sigma_std)
        if diff_std < 0.01:
            print(f"  РЕЗУЛЬТАТЫ СОВПАДАЮТ (разница: {diff_std:.6f})")
        else:
            print(f"  РАСХОЖДЕНИЕ (разница: {diff_std:.6f})")

    # Возвращаем результаты
    return {
        'x_bar': x_bar_cond,
        'D': D_cond,
        'sigma': sigma_cond
    }
raw_ph = np.array(data["pH"])
raw_sulphates = np.array(data["sulphates"])

ph_results = calculate_sample_characteristics(ph_moments, raw_ph, "pH")
sulphates_results = calculate_sample_characteristics(sulphates_moments, raw_sulphates, "SULPHATES")


    ВЫЧИСЛЕНИЕ ВЫБОРОЧНЫХ ХАРАКТЕРИСТИК ДЛЯ ПРИЗНАКА: pH

Вычисление через условные варианты
    Выборочное среднее через условные варианты: 3.3121
    Выборочная дисперсия через условные варианты: 0.0291
    Выборочное СКО: 0.1707
Вычисление по стандартной формуле
    Выборочное среднее по стандартной формуле: 3.3077
    Выборочная дисперсия по стандартной формуле: 0.0314
    Выборочное СКО по стандартной формуле: 0.1771

   СРАВНЕНИЕ РЕЗУЛЬТАТОВ

Сравнение выборочного среднего:
  РЕЗУЛЬТАТЫ СОВПАДАЮТ (разница: 0.004426)
Сравнение дисперсии:
  РЕЗУЛЬТАТЫ СОВПАДАЮТ (разница: 0.002238)
Сравнение СКО:
  РЕЗУЛЬТАТЫ СОВПАДАЮТ (разница: 0.006434)

    ВЫЧИСЛЕНИЕ ВЫБОРОЧНЫХ ХАРАКТЕРИСТИК ДЛЯ ПРИЗНАКА: SULPHATES

Вычисление через условные варианты
    Выборочное среднее через условные варианты: 0.6646
    Выборочная дисперсия через условные варианты: 0.0354
    Выборочное СКО: 0.1881
Вычисление по стандартной формуле
    Выборочное среднее по стандартной формуле: 0.6601
    Выборочная диспер

In [ ]:
def calculate_corrected_variance(sample_results, raw_data=None, feature_name=""):
    print(f"\nПризнак: {feature_name}\n")

    D_biased = sample_results['D']   # смещённая дисперсия
    sigma_biased = sample_results['sigma']
    n = len(raw_data)

    # Исправленная дисперсия
    s2 = (n / (n - 1)) * D_biased

    # Исправленное СКО
    s = np.sqrt(s2)

    print(f"Объём выборки = {n}")
    print(f"Смещённая дисперсия = {D_biased:.6f}")
    print(f"Исправленная дисперсия = {s2:.6f}")
    print()
    print(f"Смещённое СКО = {sigma_biased:.6f}")
    print(f"Исправленное СКО = {s:.6f}")

    print("\nСРАВНЕНИЕ:")
    print(f"Разница дисперсий: {s2 - D_biased:.6f}")
    print(f"Разница СКО: {s - sigma_biased:.6f}")

    return {
        's2_corrected': s2,
        's_corrected': s
    }
ph_corrected = calculate_corrected_variance(ph_results, raw_ph, "pH")
sulphates_corrected = calculate_corrected_variance(sulphates_results, raw_sulphates, "SULPHATES")


Признак: pH

Объём выборки = 114
Смещённая дисперсия = 0.029143
Исправленная дисперсия = 0.029400

Смещённое СКО = 0.170712
Исправленное СКО = 0.171466

СРАВНЕНИЕ:
Разница дисперсий: 0.000258
Разница СКО: 0.000754

Признак: SULPHATES

Объём выборки = 114
Смещённая дисперсия = 0.035382
Исправленная дисперсия = 0.035695

Смещённое СКО = 0.188101
Исправленное СКО = 0.188931

СРАВНЕНИЕ:
Разница дисперсий: 0.000313
Разница СКО: 0.000830


### 1. Вычислить точность и доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении при заданном объёме выборки для доверительной точности $\gamma \in $ {0.95, 0.99}. Сделать выводы.

In [ ]:
import scipy.stats as stats
import numpy as np
import pandas as pd
from IPython.display import display, Markdown

# Данные для pH (из предыдущих вычислений)
n_ph = len(raw_ph)
xbar_ph = ph_results['x_bar']
s_ph = ph_corrected['s_corrected']

# Данные для sulphates
n_sul = len(raw_sulphates)
xbar_sul = sulphates_results['x_bar']
s_sul = sulphates_corrected['s_corrected']

gamma_vals = [0.95, 0.99]

for feature, n, xbar, s in [("pH", n_ph, xbar_ph, s_ph), ("Sulphates", n_sul, xbar_sul, s_sul)]:
    print(f"\nПризнак: {feature}")
    for gamma in gamma_vals:
        alpha = 1 - gamma
        # двусторонний квантиль распределения Стьюдента
        t = stats.t.ppf(1 - alpha/2, df=n-1)
        delta = t * s / np.sqrt(n)
        left = xbar - delta
        right = xbar + delta
        print(f"\nγ = {gamma}:")
        print(f"  t_{gamma}({n-1}) = {t:.4f}")
        print(f"  Δ = {delta:.4f}")
        print(f"  Доверительный интервал: ({left:.4f}; {right:.4f})")
        print(f"  Интерпретация: с вероятностью {gamma} математическое ожидание")
        print(f"    генеральной совокупности находится в этих пределах.")


Признак: pH

γ = 0.95:
  t_0.95(113) = 1.9812
  Δ = 0.0318
  Доверительный интервал: (3.2803; 3.3440)
  Интерпретация: с вероятностью 0.95 математическое ожидание
    генеральной совокупности находится в этих пределах.

γ = 0.99:
  t_0.99(113) = 2.6200
  Δ = 0.0421
  Доверительный интервал: (3.2701; 3.3542)
  Интерпретация: с вероятностью 0.99 математическое ожидание
    генеральной совокупности находится в этих пределах.

Признак: Sulphates

γ = 0.95:
  t_0.95(113) = 1.9812
  Δ = 0.0351
  Доверительный интервал: (0.6296; 0.6997)
  Интерпретация: с вероятностью 0.95 математическое ожидание
    генеральной совокупности находится в этих пределах.

γ = 0.99:
  t_0.99(113) = 2.6200
  Δ = 0.0464
  Доверительный интервал: (0.6183; 0.7110)
  Интерпретация: с вероятностью 0.99 математическое ожидание
    генеральной совокупности находится в этих пределах.


Исходя из результатов вычисления точности и доверительного интервала для признаков $pH$ и $sulphates$, можно сделать вывод, что при увеличении доверительной точности $\gamma$ доверительный интервал становится шире, а точность оценки уменьшается. Это связано с тем, что для большей уверенности в покрытии истинного значения требуется больший диапазон возможных значений.

### 2.Для вычисления границ доверительного интервала для среднеквадратичного отклонения определить значение $q$ при заданных $\gamma$ и $\eta$. Построить доверительные интервалы, сделать выводы.

In [ ]:
import numpy as np
from scipy import stats

for feature, n, s in [("pH", n_ph, s_ph), ("Sulphates", n_sul, s_sul)]:
    print(f"\nПризнак: {feature}")
    print(f"  n = {n}, s = {s:.4f}")

    for gamma in gamma_vals:
        alpha = 1 - gamma

        # квантиль нормального распределения
        u = stats.norm.ppf(1 - alpha/2)

        # коэффициент q
        q = u * np.sqrt(1 / (2*(n-1)))

        # границы доверительного интервала
        lower = s * (1 - q)
        upper = s * (1 + q)

        print(f"\nγ = {gamma}:")
        print(f"  uγ = Φ⁻¹(1-α/2) = Φ⁻¹({1-alpha/2:.3f}) = {u:.4f}")
        print(f"  q = uγ·√(1/(2(n-1))) = {u:.4f}·√(1/(2·{n-1})) = {q:.4f}")
        print(f"  Нижняя граница: σ_н = s(1-q) = {s:.4f}·(1-{q:.4f}) = {lower:.4f}")
        print(f"  Верхняя граница: σ_в = s(1+q) = {s:.4f}·(1+{q:.4f}) = {upper:.4f}")


Признак: pH
  n = 114, s = 0.1715

γ = 0.95:
  uγ = Φ⁻¹(1-α/2) = Φ⁻¹(0.975) = 1.9600
  q = uγ·√(1/(2(n-1))) = 1.9600·√(1/(2·113)) = 0.1304
  Нижняя граница: σ_н = s(1-q) = 0.1715·(1-0.1304) = 0.1491
  Верхняя граница: σ_в = s(1+q) = 0.1715·(1+0.1304) = 0.1938

γ = 0.99:
  uγ = Φ⁻¹(1-α/2) = Φ⁻¹(0.995) = 2.5758
  q = uγ·√(1/(2(n-1))) = 2.5758·√(1/(2·113)) = 0.1713
  Нижняя граница: σ_н = s(1-q) = 0.1715·(1-0.1713) = 0.1421
  Верхняя граница: σ_в = s(1+q) = 0.1715·(1+0.1713) = 0.2008

Признак: Sulphates
  n = 114, s = 0.1889

γ = 0.95:
  uγ = Φ⁻¹(1-α/2) = Φ⁻¹(0.975) = 1.9600
  q = uγ·√(1/(2(n-1))) = 1.9600·√(1/(2·113)) = 0.1304
  Нижняя граница: σ_н = s(1-q) = 0.1889·(1-0.1304) = 0.1643
  Верхняя граница: σ_в = s(1+q) = 0.1889·(1+0.1304) = 0.2136

γ = 0.99:
  uγ = Φ⁻¹(1-α/2) = Φ⁻¹(0.995) = 2.5758
  q = uγ·√(1/(2(n-1))) = 2.5758·√(1/(2·113)) = 0.1713
  Нижняя граница: σ_н = s(1-q) = 0.1889·(1-0.1713) = 0.1566
  Верхняя граница: σ_в = s(1+q) = 0.1889·(1+0.1713) = 0.2213


Для обоих признаков были построены доверительные интервалы для истинного стандартного отклонения σ генеральной совокупности.

Для признака $pH$:

при $\gamma = 0.95: \sigma \in [0.1491; 0.1938]$

при $\gamma = 0.99: \sigma \in [0.1421; 0.2008]$

Полученные значения показывают, что разброс значений $pH$ относительно среднего невелик.

Для признака $sulphates$:

при $\gamma = 0.95: \sigma \in [0.1643; 0.2136]$

при $\gamma = 0.99: \sigma \in [0.1566; 0.2213]$

Это говорит о том, что разброс значений $sulphates$ немного больше, чем у $pH$.

Исходя из результатов, получено, что при увеличении уровня доверия интервал расширяется. Следовательно, можно сделать вывод, что с заданными уровнями доверия истинные значения стандартных отклонений признаков $pH$ и $sulphates$ находятся в соответствующих найденных диапазонах.

### 3. Проверить гипотезу о нормальности заданного распределения с помощью критерия $\chi^2$ (Пирсона). Для этого необходимо найти теоретические частоты и вычислить наблюдаемое значение критерия. Для удобства вычисления необходимо заполнить табл. 1.

In [ ]:
def chi2_goodness_of_fit(interval_table, raw_data, xbar, s, feature_name):
    print(f"\nПроверка для признака: {feature_name}")
    print(f"  x̄ = {xbar:.4f}, s = {s:.4f}")

    df = interval_table.iloc[:-1].copy()
    n_total = len(raw_data)

    left_bounds = []
    right_bounds = []
    intervals = []

    for val in df["$[x_i, x_{i+1})$"]:
        l = float(val.split(',')[0].strip('[ '))
        r = float(val.split(',')[1].strip(' )]'))
        left_bounds.append(l)
        right_bounds.append(r)
        intervals.append(f"[{l:.3f}, {r:.3f})")

    n_i = df["$m_i$"].values.astype(int)
    k = len(n_i)

    # Вычисляем теоретические вероятности и частоты
    p_i = []
    n_i_prime = []

    for i in range(k):
        if i == 0:
            # Для первого интервала
            prob = stats.norm.cdf(right_bounds[i], xbar, s)
        elif i == k-1:
            # Для последнего интервала
            prob = 1 - stats.norm.cdf(left_bounds[i], xbar, s)
        else:
            prob = stats.norm.cdf(right_bounds[i], xbar, s) - stats.norm.cdf(left_bounds[i], xbar, s)

        p_i.append(prob)
        n_i_prime.append(prob * n_total)

    table_data = []
    sum_n_i_sq_div = 0

    for i in range(k):
        n_i_sq_div = n_i[i]**2 / n_i_prime[i] if n_i_prime[i] > 0 else 0
        sum_n_i_sq_div += n_i_sq_div

        table_data.append({
            "i": i+1,
            "$[x_i, x_{i+1})$": intervals[i],
            "$n_i$": n_i[i],
            "$p_i$": f"{p_i[i]:.4f}",
            "$n'_i$": f"{n_i_prime[i]:.2f}",
            "$(n_i - n'_i)^2$": f"{(n_i[i] - n_i_prime[i])**2:.2f}",
            "$\\frac{(n_i - n'_i)^2}{n'_i}$": f"{(n_i[i] - n_i_prime[i])**2 / n_i_prime[i]:.4f}" if n_i_prime[i] > 0 else "-",
            "$n_i^2$": n_i[i]**2,
            "$\\frac{n_i^2}{n'_i}$": f"{n_i_sq_div:.2f}"
        })

    chi2_obs = sum(float(row["$\\frac{(n_i - n'_i)^2}{n'_i}$"]) for row in table_data if row["$\\frac{(n_i - n'_i)^2}{n'_i}$"] != "-")

    table_data.append({
        "i": "Σ",
        "$[x_i, x_{i+1})$": "-",
        "$n_i$": sum(n_i),
        "$p_i$": "-",
        "$n'_i$": f"{sum(n_i_prime):.2f}",
        "$(n_i - n'_i)^2$": "-",
        "$\\frac{(n_i - n'_i)^2}{n'_i}$": f"{chi2_obs:.4f}",
        "$n_i^2$": "-",
        "$\\frac{n_i^2}{n'_i}$": f"{sum_n_i_sq_div:.2f}"
    })

    result_table = pd.DataFrame(table_data)
    display(Markdown(result_table.to_markdown(index=False)))

    print(f"\nχ²_набл = {chi2_obs:.4f}")
    return result_table, chi2_obs


ph_chi2_goodness_of_fit, chi_obs_ph = chi2_goodness_of_fit(ph_table, raw_ph, ph_results['x_bar'], ph_corrected['s_corrected'], "pH")
sulphates_chi2_goodness_of_fit, chi_obs_sulphates = chi2_goodness_of_fit(sulphates_table, raw_sulphates, sulphates_results['x_bar'], sulphates_corrected['s_corrected'], "SULPHATES")


Проверка для признака: pH
  x̄ = 3.3121, s = 0.1715


| i   | $[x_i, x_{i+1})$   |   $n_i$ | $p_i$   |   $n'_i$ | $(n_i - n'_i)^2$   |   $\frac{(n_i - n'_i)^2}{n'_i}$ | $n_i^2$   |   $\frac{n_i^2}{n'_i}$ |
|:----|:-------------------|--------:|:--------|---------:|:-------------------|--------------------------------:|:----------|-----------------------:|
| 1   | [2.860, 3.010)     |       5 | 0.0390  |     4.45 | 0.30               |                          0.0683 | 25        |                   5.62 |
| 2   | [3.010, 3.160)     |      12 | 0.1484  |    16.92 | 24.22              |                          1.4311 | 144       |                   8.51 |
| 3   | [3.160, 3.310)     |      38 | 0.3076  |    35.06 | 8.64               |                          0.2463 | 1444      |                  41.18 |
| 4   | [3.310, 3.450)     |      41 | 0.2943  |    33.55 | 55.52              |                          1.655  | 1681      |                  50.11 |
| 5   | [3.450, 3.600)     |      13 | 0.1641  |    18.71 | 32.58              |                          1.7417 | 169       |                   9.03 |
| 6   | [3.600, 3.750)     |       3 | 0.0413  |     4.7  | 2.90               |                          0.6174 | 9         |                   1.91 |
| 7   | [3.750, 3.900)     |       2 | 0.0053  |     0.61 | 1.94               |                          3.1899 | 4         |                   6.58 |
| Σ   | -                  |     114 | -       |   114    | -                  |                          8.9497 | -         |                 122.95 |


χ²_набл = 8.9497

Проверка для признака: SULPHATES
  x̄ = 0.6646, s = 0.1889


| i   | $[x_i, x_{i+1})$   |   $n_i$ | $p_i$   |   $n'_i$ | $(n_i - n'_i)^2$   |   $\frac{(n_i - n'_i)^2}{n'_i}$ | $n_i^2$   |   $\frac{n_i^2}{n'_i}$ |
|:----|:-------------------|--------:|:--------|---------:|:-------------------|--------------------------------:|:----------|-----------------------:|
| 1   | [0.440, 0.610)     |      53 | 0.3862  |    44.03 | 80.48              |                          1.828  | 2809      |                  63.8  |
| 2   | [0.610, 0.780)     |      42 | 0.3431  |    39.11 | 8.36               |                          0.2139 | 1764      |                  45.11 |
| 3   | [0.780, 0.950)     |      13 | 0.2053  |    23.4  | 108.15             |                          4.622  | 169       |                   7.22 |
| 4   | [0.950, 1.110)     |       1 | 0.0563  |     6.41 | 29.31              |                          4.5702 | 1         |                   0.16 |
| 5   | [1.110, 1.280)     |       2 | 0.0086  |     0.99 | 1.03               |                          1.0453 | 4         |                   4.06 |
| 6   | [1.280, 1.450)     |       2 | 0.0005  |     0.06 | 3.75               |                         60.2424 | 4         |                  64.18 |
| 7   | [1.450, 1.620)     |       1 | 0.0000  |     0    | 1.00               |                        541.768  | 1         |                 543.77 |
| Σ   | -                  |     114 | -       |   114    | -                  |                        614.289  | -         |                 728.29 |


χ²_набл = 614.2894


### 4. Доказать, что $\chi^2_{набл} = \sum_{i=1}^{k} \frac{n^2_i}{n'_i} - n$. Проконтролировать корректность вычисления $\chi^2_{набл}$

Из определения

$$\chi^2_{набл} = \sum_{i=1}^{k} \frac{(n_i - n'_i)^2}{n'_i} $$

Выполним математические преобразования:

$ \sum_{i=1}^{k} \frac{(n_i - n'_i)^2}{n'_i} = \sum_{i=1}^{k} \frac{n_i^2 -2n_i n'_i + n_i'^2}{n'_i}= \sum_{i=1}^{k} \frac{n_i^2}{n'_i} - 2\sum_{i=1}^{k} n_i + \sum_{i=1}^{k} n'_i = \sum_{i=1}^{k} \frac{n_i^2}{n'_i} -2n + n = \sum_{i=1}^{k} \frac{n^2_i}{n'_i} - n$

Таким образом,

$$\chi^2_{набл} = \sum_{i=1}^{k} \frac{n^2_i}{n'_i} - n $$


In [ ]:
def check_correct_chi2(result, feature_name, chi_2):
  print(f"\nПроверка для признака: {feature_name}")
  n_total = 114
  sum_n_i = sum(list(map(float, result["$\\frac{n_i^2}{n'_i}$"][:len(result)-1])))

  print(f"\nИсходная формула: χ²_набл = {chi_2:.2f}")
  chi_2_simple = sum_n_i - n_total
  print(f"\nВыведенная формула: χ²_набл = {chi_2_simple:.2f}")


check_correct_chi2(ph_chi2_goodness_of_fit, "pH", chi_obs_ph)
check_correct_chi2(sulphates_chi2_goodness_of_fit, "SULPHATES", chi_obs_sulphates)


Проверка для признака: pH

Исходная формула: χ²_набл = 8.95

Выведенная формула: χ²_набл = 8.94

Проверка для признака: SULPHATES

Исходная формула: χ²_набл = 614.29

Выведенная формула: χ²_набл = 614.30


Таким образом, получено, что формула для нахождения $\chi^2_{набл}$ верна и проконтролирована корректность вычисления критерия. Небольшие расхождения объясняются округлением частот при вычислении.

### 5. По заданному уровню значимости $\alpha = 0.05$ и числу степеней свободы $d f$ найти критическую точку $\chi^2_{крит}$ и сравнить с наблюдаемым значением. Сделать выводы.

In [ ]:
def chi2(interval_table, raw_data, xbar, s, feature_name):
    print(f"\nПроверка для признака: {feature_name}")
    print(f"  x̄ = {xbar:.4f}, s = {s:.4f}")
    print(f"  Уровень значимости α = 0.05")

    df = interval_table.iloc[:-1].copy()
    n_total = len(raw_data)

    left_bounds = []
    right_bounds = []
    intervals = []

    for val in df["$[x_i, x_{i+1})$"]:
        l = float(val.split(',')[0].strip('[ '))
        r = float(val.split(',')[1].strip(' )]'))
        left_bounds.append(l)
        right_bounds.append(r)
        intervals.append(f"[{l:.3f}, {r:.3f})")

    n_i = df["$m_i$"].values.astype(int)
    k = len(n_i)

    # Вычисляем теоретические вероятности и частоты
    p_i = []
    n_i_prime = []

    for i in range(k):
        if i == 0:
            # Для первого интервала
            prob = stats.norm.cdf(right_bounds[i], xbar, s)
        elif i == k-1:
            # Для последнего интервала
            prob = 1 - stats.norm.cdf(left_bounds[i], xbar, s)
        else:
            prob = stats.norm.cdf(right_bounds[i], xbar, s) - stats.norm.cdf(left_bounds[i], xbar, s)

        p_i.append(prob)
        n_i_prime.append(prob * n_total)

    table_data = []
    sum_n_i_sq_div = 0

    for i in range(k):
        n_i_sq_div = n_i[i]**2 / n_i_prime[i] if n_i_prime[i] > 0 else 0
        sum_n_i_sq_div += n_i_sq_div

        table_data.append({
            "i": i+1,
            "$[x_i, x_{i+1})$": intervals[i],
            "$n_i$": n_i[i],
            "$p_i$": f"{p_i[i]:.4f}",
            "$n'_i$": f"{n_i_prime[i]:.2f}",
            "$(n_i - n'_i)^2$": f"{(n_i[i] - n_i_prime[i])**2:.2f}",
            "$\\frac{(n_i - n'_i)^2}{n'_i}$": f"{(n_i[i] - n_i_prime[i])**2 / n_i_prime[i]:.4f}" if n_i_prime[i] > 0 else "-",
            "$n_i^2$": n_i[i]**2,
            "$\\frac{n_i^2}{n'_i}$": f"{n_i_sq_div:.2f}"
        })

    chi2_obs = sum(float(row["$\\frac{(n_i - n'_i)^2}{n'_i}$"]) for row in table_data if row["$\\frac{(n_i - n'_i)^2}{n'_i}$"] != "-")

    chi2_simple = sum_n_i_sq_div - n_total
    print(f"\nχ²_набл = {chi2_obs:.4f}")

    # Критическое значение при α = 0.05
    df = k - 3
    chi2_crit = stats.chi2.ppf(0.95, df)

    print(f"\nЧисло степеней свободы: df = k - 3 = {k} - 3 = {df}")
    print(f"Критическое значение χ²_крит = {chi2_crit:.4f}")


chi2(ph_table, raw_ph, ph_results['x_bar'], ph_corrected['s_corrected'], "pH")
chi2(sulphates_table, raw_sulphates, sulphates_results['x_bar'], sulphates_corrected['s_corrected'], "SULPHATES")


Проверка для признака: pH
  x̄ = 3.3121, s = 0.1715
  Уровень значимости α = 0.05

χ²_набл = 8.9497

Число степеней свободы: df = k - 3 = 7 - 3 = 4
Критическое значение χ²_крит = 9.4877

Проверка для признака: SULPHATES
  x̄ = 0.6646, s = 0.1889
  Уровень значимости α = 0.05

χ²_набл = 614.2894

Число степеней свободы: df = k - 3 = 7 - 3 = 4
Критическое значение χ²_крит = 9.4877


Для проверки гипотезы о нормальности распределения необходимо проверить выполнение неравенства:
$$\chi^2_{набл} \leq \chi^2_{крит}$$

Исходя из полученных результатов для признаков:

Для признака $pH$:
$$\chi^2_{набл} = 8.9497$$
$$\chi^2_{крит} = 9.4877$$
$$\chi^2_{набл} \leq \chi^2_{крит}$$

Таким образом, гипотеза принимается и, следовательно, распределение признака $pH$ можно считать нормальным на уровне значимости 0.05.


Для признака $sulphates$:
$$\chi^2_{набл} = 614.2894$$
$$\chi^2_{крит} = 9.4877$$
$$\chi^2_{набл} > \chi^2_{крит}$$

Таким образом, гипотеза отвергается и, следовательно, распределение признака $sulphates$ не является нормальным.


### Вывод

В ходе работы выполнены задачи по вычислению интервальных оценок параметров распределения и проверке гипотезы о нормальности выборочных данных.

Для признаков $pH$ и $sulphates$ вычислены доверительные интервалы для математического ожидания при неизвестном среднеквадратичном отклонении. Интервалы построены при уровнях доверия $ \gamma = 0.95$ и $ \gamma = 0.99$. Результаты показали, что с вероятностью $95\%$ или $99\%$ истинное значение математического ожидания находится внутри указанных границ. Увеличение уровня доверия приводит к расширению интервала, что соответствует более высокой вероятности попадания истинного значения.

С использованием приближённой формулы через коэффициент $q$ построены доверительные интервалы для $\sigma$. Увеличение $\gamma$ также приводит к расширению интервала.

Используя критерий $\chi^2$ Пирсона, вычислены теоретические частоты, наблюдаемое значение $\chi^2$ и проведено сравнение с критической точкой при $\alpha = 0.05$. Полученные результаты позволили сделать вывод о том, что распределение признака $pH$ можно считать нормальным (гипотеза принимается), для признака $sulphates$ гипотеза отвергается.